pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -7,22 +7,35 @@ import (
 	"strings"
 	"time"
 
-	"github.com/grafana/grafana/pkg/services/folder"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/store"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
+type RuleLimits struct {
+	// The default interval if not specified.
+	DefaultRuleEvaluationInterval time.Duration
+	// All intervals must be an integer multiple of this duration.
+	BaseInterval time.Duration
+}
+
+func RuleLimitsFromConfig(cfg *setting.UnifiedAlertingSettings) RuleLimits {
+	return RuleLimits{
+		DefaultRuleEvaluationInterval: cfg.DefaultRuleEvaluationInterval,
+		BaseInterval:                  cfg.BaseInterval,
+	}
+}
+
 // validateRuleNode validates API model (definitions.PostableExtendedRuleNode) and converts it to models.AlertRule
 func validateRuleNode(
 	ruleNode *apimodels.PostableExtendedRuleNode,
 	groupName string,
 	interval time.Duration,
 	orgId int64,
-	namespace *folder.Folder,
-	cfg *setting.UnifiedAlertingSettings) (*ngmodels.AlertRule, error) {
-	intervalSeconds, err := validateInterval(cfg, interval)
+	namespaceUID string,
```

Comment:
Debated on this... we could keep folder if we really really need, that one isn't as egregious as the unified alerting settings. We are trading type safety for simplicity here.